src/main/java/com/thirdcc/webapp/service/impl/TransactionServiceImpl.java

```diff
@@ -168,22 +176,33 @@ public EventBudgetTotalDTO findTotalTransactionByEventId(Long eventId) {
         return eventBudgetTotalDTO;
     }
 
-    private TransactionDTO mapEventName(TransactionDTO transactionDTO) {
-        Event event = eventRepository.findById(transactionDTO.getEventId())
-            .orElseThrow(() -> new BadRequestException("Event Not Found "+ transactionDTO.getEventId()));
-        transactionDTO.setEventName(event.getName());
-        return transactionDTO;
-    }
-
-    private ReceiptDTO saveReceipt(ReceiptDTO receiptDTO) {
-        receiptDTO = uploadReceipt(receiptDTO);
-        return receiptService.save(receiptDTO);
-    }
-
-    private ReceiptDTO uploadReceipt(ReceiptDTO receiptDTO) {
-        // TODO: upload receipt image to cloud storage
-        // TODO: set info of uploaded receipt to receiptDTO
-        return receiptDTO;
+    /**
+     * Validate on fields:
+     * - title
+     * - transaction_amount
+     * - transaction_status
+     * - file
+     *
+     * @param transactionDTO the TransactionDTO to validate.
+     * @return is the TransactionDTO valid
+     */
+    private boolean validateTransactionDTO(TransactionDTO transactionDTO, MultipartFile multipartFile){
+
+        boolean hasTitle = !transactionDTO.getTitle().isEmpty();
+        boolean hasTransactionAmount = transactionDTO.getTransactionAmount() != null && transactionDTO.getTransactionAmount().intValue() >= 0;
+        boolean hasTransactionStatus = transactionDTO.getTransactionStatus() != null;
+        boolean hasImageLink = (transactionDTO.getImageLink() != null && !transactionDTO.getImageLink().isEmpty());
+
+        if (transactionDTO.getTransactionType() == TransactionType.EXPENSE){
+            // if there is no image link, then it is a new transaction, thus must have an image file to be uploaded
+            if (!hasImageLink){
+                if (multipartFile == null || multipartFile.isEmpty()) return false;
+            }
+            return (hasTitle && hasImageLink && hasTransactionAmount && hasTransactionStatus);
+        }else{
+            // if this is INCOME (DEBT), then image link is not necessary, because treasurer doesn't have to audit this transaction with receipt image
+            return (hasTitle && hasTransactionAmount && hasTransactionStatus);
+        }
```

Comment:
If I updating a Transaction in my request data, I must have the following data:
- imageLink
- transaction Info (amount, title, status)

But If I updating with new MultipartFile and no ImageLink, 
then the Receipt Image will be replaced right?

Reply:
👌